docs/source/examples/classification_and_regression.rst

```diff
@@ -0,0 +1,324 @@
+            
+            # create cnn model
+            model = models.__dict__[model_name](True)
+            
+            # remove fc layers and add a new fc layer
+            num_features = model.fc.in_features
+            model.fc = nn.Linear(num_features, 6) # classifier + localizer
+            self.model = model
+        
+        def forward(self, x):
+            x = self.model(x)                    # extract features from CNN
+            scores = x[:, :self.num_classes]     # class scores
+            coords = x[:, self.num_classes:]     # coordinates
+            return [scores, torch.sigmoid(coords)]   # sigmoid output is in the range of [0, 1]
+
+Regarding the complexity of the problem, the number of the samples in the training dataset, and the similarity of the training dataset to the ImageNet dataset, we may decide to freeze some of the layers. In our current example, based on the mentioned factors, we freeze just the last fully connected layer.
+
+.. code-block:: python
+
+    network = ClassifierLocalizer(model_name='resnet18')
+
+    def freeze_weights(network):
+        for name, param in network.named_parameters():
+            if not name.startswith('fc.'):
+                param.requires_grad = False
+
+    freeze_weights(network)
+    print(network)
+
+.. code-block:: python
+
+    network = ClassifierLocalizer(model_name='resnet18')  # network without freezing any layer.
+
+Loss function
+=============
+
+As we discussed before, we have two different tasks in this example. These tasks need different loss functions; Cross-Entropy loss for the classification and Mean Square Error loss for the regression. Below, we define a new loss function class that sums both losses to considers them simultaneously. However, as the regression is relatively a harder task, we apply a higher weight to MSEloss.
+
+.. code-block:: python
+
+    class classification_regression_Loss(nn.Module):
+        def __init__(self):
+            super(classification_regression_Loss, self).__init__()
+            self.ce_loss = nn.CrossEntropyLoss() # size_average=False
+            self.mse_loss = nn.MSELoss()
+            
+        def forward(self, y_pred, y_true):
+            loss_cls = self.ce_loss(y_pred[0], y_true[0]) # Cross Entropy Error (for classification)
+            loss_reg = self.mse_loss(y_pred[1], y_true[ 1]) # Mean Squared Error (for landmarks)
+            total_loss = loss_reg + loss_cls
+            return total_loss
+
+Training
+========
+
+.. code-block:: python
+
+    optimizer = optim.Adam(network.parameters(), lr=0.0001, weight_decay=0)
+    loss_function = classification_regression_Loss()
+    #Step_Learning_Rate = StepLR(step_size=2 , gamma=0.1, last_epoch=-1, verbose=False)
+    exp = Experiment('./two_task_example', network, optimizer=optimizer, loss_function=loss_function, device="all")
+    exp.train(train_dataloader, valid_dataloader, callbacks=callbacks, epochs=num_epochs)
```

Comment:
There is no `callbacks` defined above. Also, the ones you defined in the notebook are not useful since you are using Experiment and Experiment already has all of them.

Reply:
I agree with you. however in the code:
**exp.train(train_dataloader, valid_dataloader, callbacks=callbacks, epochs=num_epochs)**
if i want to use the callbacks argument, i should define it first. isn't it?

Reply:
I don't understand, what callbacks do you want to use?

Reply:
Solved